### PR TITLE
Initial integrate our L0 Llama impl into ipex-llm

### PR DESCRIPTION
**Pull request by rnwang04**
### 1. Why the change?

To integrate our current L0 implementation to ipex-llm: https://github.com/analytics-zoo/nano/issues/1687
Need to merge with https://github.com/[intel-analytics/llm.cpp](https://github.com/intel-analytics/llm.cpp/pull/593)/pull/593

### 2. User API changes

See pipeline `llama.py` example here: https://github.com/intel-analytics/ipex-llm/blob/08b831428fdc1c7708cce99c99857b064cdbb7e9/python/llm/example/NPU/HF-Transformers-AutoModels/LLM/Pipeline-Models/llama.py

### 3. Summary of the change 

- Support the convert process from HF model to blob for Llama2
- Adapted according to our latest interface
- Code refactor

### 4. How to test?

- [x] Unit test: Please manually trigger the PR Validation [here](https://github.com/intel-analytics/ipex-llm-workflow/actions/workflows/llm-PR-validation.yml) by inputting the PR number (e.g., `1234`). And paste your action link here once it has been successfully finished.
- [x] Application test

### 5. Unsolved issues (maybe in next PR)

- [ ] save & load
- [ ] more checks

### CODE REVIEW

**Comment by rnwang04** [comment]
PR validation is passed: https://github.com/intel-analytics/ipex-llm-workflow/actions/runs/11478332409

File: python/llm/src/ipex_llm/transformers/npu_model.py [hunk 1]
```diff
@@ -120,6 +120,7 @@ def from_pretrained(cls, *args, **kwargs):
         ignore_argument(kwargs, "speculative")
         ignore_argument(kwargs, "pipeline_parallel_stages")
         optimize_model = kwargs.pop("optimize_model", False)
+        pipeline = kwargs.pop("pipeline", True)
```
Comment (jason-dai) [inline python/llm/src/ipex_llm/transformers/npu_model.py:123]: default to false?

Comment (rnwang04) [reply python/llm/src/ipex_llm/transformers/npu_model.py]: Done

**Review by jason-dai — APPROVED** [review]
LGTM

**Review by plusbang — APPROVED** [review]
LGTM

**Pull request merged**